py/server/deephaven/dbc.py

```diff
@@ -0,0 +1,48 @@
+#
+#     Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+#
+"""This module supports ingesting data from external databases into Deephaven via database connectivity interfaces.
+
+Currently, it requires that application must use Turbodbc (https://turbodbc.readthedocs.io/en/latest/index.html) to
+access a supported relational database. Turbodbc has built-in Apache Arrow support which enables Deephaven to
+achieve maximum performance and efficiency when ingesting relational data.
+
+Looking ahead, support of Arrow Database Connectivity (ADBC) https://arrow.apache.org/docs/dev/format/ADBC.html
+will be added once it reaches stable status. ADBC offers a standard API to fetch data in Arrow format from data
+sources that support Arrow natively as well as ODBC-only databases.
+"""
+from typing import Any
+
+from deephaven.table import Table
+from deephaven import arrow as dharrow
+from deephaven import DHError
+
+try:
+    import turbodbc
+except ImportError:
+    raise DHError(message="import turbodbc failed")
+
+
+def read_cursor(cursor: Any) -> Table:
+    """Converts the result set of the provided cursor into a Deephaven table.
+
+    Args:
+        cursor (Any): a Turbodbc cursor. Prior to it being passed in, its execute() method must be called to run a query
+            operation that produces a result set
```

Comment:
turbodbc is the only Python module that supports fetching ODBC data into arrow tables and since we don't want to process the ODBC result set ourselves, we require the user to use turbodbc.